EIPS/eip-6206.md

```diff
@@ -52,26 +52,26 @@ Let the definition of `type[i]` be inherited from [EIP-4750](./eip-4750.md) and
   * `JUMPF` into non-returning section (`type[target_section_index].outputs` equals `0x80`): `stack_height_min` MUST be greater than or equal to `type[target_section_index].inputs`.
 * Stack overflow check at `JUMPF`: `stack_height_max` MUST be less than or equal to `1024 - types[target_section_index].max_stack_height + types[target_section_index].inputs`.
 * `JUMPF` is considered terminating instruction, i.e. does not have successor instructions in code validation and MAY be final instruction in the section. 
 * The code validation defined in [EIP-4200](./eip-4200.md) also fails if any `RJUMP*` offset points to one of the two bytes directly following a `JUMPF` instruction.
 
 `CALLF` instruction validation is extended to include the rule:
 
-* Code section is invalid in case an immediate argument `target_section_index` of any `CALLF` targets a non-returning section, i.e. `type[target_section_index]` equals `0x80`.
+* Code section is invalid in case an immediate argument `target_section_index` of any `CALLF` targets a non-returning section, i.e. `type[target_section_index].outputs` equals `0x80`.
 
 #### Non-returning status validation
 
-Section type MUST be non-returning in case the section contains no `RETF` instructions and no `JUMPF` instructions targeting returning sections (target section's status is checked via its output value in type section.)
+Section type MUST be non-returning if and only if the section contains no `RETF` instructions and no `JUMPF` instructions targeting returning sections (target section's status is checked via its output value in type section.)
 *Note: This implies that section containing only `JUMPF`s into non-returning sections is non-returning itself.*
 
 ## Rationale
 
 ### Allowing `JUMPF` to section with less outputs
 
-As long as `JUMPF` prepares the delta `type[current_section_index].outputs - type[target_section_index].outputs` stack elements before changing code sections, it is possible to jump to a section with less outputs than was originally entered via `CALLF`. This will reduce duplicated code as it will allow compilers more flexibility during code generation such that certain helpers can be used generically by functions, regardless of their output values.
+A code section with a `JUMPF` can, just before that instruction, have a delta of `type[current_section_index].outputs - type[target_section_index].outputs` output elements on the operand stack (i.e. counted on top of the target section inputs!). This makes it possible to `JUMPF` to a section with less outputs than those of the current section. This will reduce duplicated code as it will allow compilers more flexibility during code generation such that certain helpers can be used generically by functions, regardless of their output values.
```

Comment:
> (i.e. counted on top of the target section inputs!).

This is quite confusing, output elements are _below_ target section inputs, right? I would just remove the part in parenthesis.